### PR TITLE
utils: Expand capability beyond u8, int to ASCII

### PR DESCRIPTION
**Pull request by emz229**
This merge will allow users more flexibility to convert between integers and ASCII types for use with various layers such as CFB.

### CODE REVIEW

**Comment by emz229** [comment]
@andyross @nashif Now that all checks have passed, I believe this is ready for review.

File: lib/utils/dec.c [hunk 1]
```diff
@@ -3,31 +3,49 @@
...
+int itoa_util(int value, char *buf, size_t buflen, int base)
+{
+	size_t num_digits = 0;
+	char buffer[sizeof(value) * CHAR_BIT + 2];
+	static const char digits[37] = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+
+	/* Check for valid base */
+	if (base < 2 || base > 36) {
 		*buf = '\0';
 	}
```
Comment (peter-mitsis) [inline lib/utils/dec.c:25]: Two items.

1. If base is out of range, I we ought to return early.
2. There also ought to be a check for buf == NULL.

Comment (emz229) [reply lib/utils/dec.c]: excellent points! Changes have been made

**Review by peter-mitsis — APPROVED** [review]

**Comment by emz229** [comment]
@peter-mitsis Do you know if any other maintainers are available to review this PR?

**Pull request closed without merging**